src/ansys/fluent/interface/post_objects/post_object_definitions.py

```diff
@@ -432,15 +432,27 @@ def value(self):
             auto_range_off = self.get_ancestors_by_type(
                 ContourDefn
             ).range.auto_range_off
-            if not filled or (auto_range_off and auto_range_off.clip_to_range()):
+            if self._value is False and (
+                not filled or (auto_range_off and auto_range_off.clip_to_range())
+            ):
                 logger.warning(
-                    "For unfilled and clipped contours node values are displayed."
+                    "For unfilled and clipped contours, node values must be displayed. "
                 )
                 self._value = True
```

Comment:
@prmukherj 
- This existing side-effect in the getter looks like a definite defect (`self._value = True`). Is there any problem if we just get rid of that?

Reply:
I think this a py_viz level modification as in fluent there is no such limitation:
![image](https://github.com/user-attachments/assets/dcfb970b-d923-4cec-856b-26487521b43c)

Will just update the code to return True and not update the value as discussed.

Reply:
A better check is in place now, please look at the tests for more details. This condition has been checked for all the dependent parameters.